### DISCUSSION TITLE
Column Size

### DISCUSSION THREAD

**Discussion by Rucchao** [duckdb/duckdb#5057] [General]
Hi,

I am curious about how to get the column size as per each column in DuckDB, currently only the database_size can be obtained.

A follow-up question is how to get the encoding type of each column. Any comments would be appreciated.

Thanks!

**Comment by Mytherin** [comment]
SELECT * FROM pragma_storage_info(‘tbl’) should work

**Reply by stas-sl** [reply]
@Mytherin, is it really possible to get compressed/uncompressed total column sizes from `pragma_storage_info`? It contains information about compression used for each row group/column, but it is not obvious for me how to get columns sizes from it.
